src/test/java/org/apache/commons/compress/compressors/z/ZCompressorInputStreamTest.java

```diff
@@ -68,4 +73,32 @@ public void testSingleByteReadConsistentlyReturnsMinusOneAtEof() throws IOExcept
         }
     }
 
+    @Test
+    public void testInvalidMaxCodeSize() throws IOException {
+        Set<Integer> invalidValues = new TreeSet<>();
+        invalidValues.addAll(IntStream.range(Byte.MIN_VALUE, -120).boxed().collect(Collectors.toSet()));
+        invalidValues.addAll(IntStream.range(-97, -88).boxed().collect(Collectors.toSet()));
+        invalidValues.addAll(IntStream.range(-65, -56).boxed().collect(Collectors.toSet()));
+        invalidValues.addAll(IntStream.range(-33, -24).boxed().collect(Collectors.toSet()));
+        invalidValues.addAll(IntStream.range(-1, 8).boxed().collect(Collectors.toSet()));
+        invalidValues.addAll(IntStream.range(31, 40).boxed().collect(Collectors.toSet()));
+        invalidValues.addAll(IntStream.range(63, 72).boxed().collect(Collectors.toSet()));
+        invalidValues.addAll(IntStream.range(95, 104).boxed().collect(Collectors.toSet()));
+        invalidValues.add(127);
+
+        final File input = getFile("bla.tar.Z");
+        try (final InputStream contentStream = Files.newInputStream(input.toPath())) {
+            final byte[] content = IOUtils.toByteArray(contentStream);
+
+            for (int value : invalidValues) {
+                content[2] = (byte) value;
+
+                // Test that invalid values always throw an IOException
+                assertThrows(IOException.class, () ->
+                        new ZCompressorInputStream(new ByteArrayInputStream(content), 1024 * 1024)
+                );
+            }
+        }
```

Comment:
Simplify and clarify: You can read the file test fixture's content in one call. You can just stream over the IntStreams.
```
        final byte[] bytes = Files.readAllBytes(AbstractTest.getPath("bla.tar.Z"));

        // @formatter:off
        final IntStream[] invalid = {
            IntStream.range(Byte.MIN_VALUE, -120),
            IntStream.range(-97, -88),
            IntStream.range(-65, -56),
            IntStream.range(-33, -24),
            IntStream.range(-1, 8),
            IntStream.range(31, 40),
            IntStream.range(63, 72),
            IntStream.range(95, 104),
            IntStream.range(127, 127)
            };
        // @formatter:on

        Stream.of(invalid).forEach(ints -> ints.forEach(i -> {
            bytes[2] = (byte) i;
            assertThrows(IllegalArgumentException.class, () -> new ZCompressorInputStream(new ByteArrayInputStream(bytes), 1024 * 1024),
                    () -> "value=" + i);
        }));
```
Also, the exception should be thrown in the initializeTables method where the check is already performed and can be expanded.

Reply:
Many methods throw unchecked exceptions in this library and in most libraries and IN the JDK, so any truly billet-proof app will know where it needs to watch out for this based on testing, fuzzing, and general empirical usage. All of this to say that the method ALREADY threw IAE, so now it does so under another use case. IOW, it's fine the way it is IMO. I'm happy to hear other opinions.